newIDE/app/src/ResourcesList/ResourceThumbnail/index.js

```diff
@@ -15,61 +15,50 @@ type Props = {|
   onContextMenu?: (number, number) => void,
 |};
 
-type State = {|
-  resourceKind: ?ResourceKind,
-|};
-
 /**
  * Display the right thumbnail for any given resource of a project
  */
-export default class ResourceThumbnail extends React.Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-
-    this.state = this._loadFrom(props);
-  }
+const ResourceThumbnail = ({
+  project,
+  resourceName,
+  resourcesLoader,
+  style,
+  selectable,
+  selected,
+  onSelect,
+  onContextMenu,
+}: Props) => {
+  const [resourceKind, setResourceKind] = React.useState<?ResourceKind>(null);
 
-  // To be updated, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops.
-  UNSAFE_componentWillReceiveProps(newProps: Props) {
-    if (
-      newProps.resourceName !== this.props.resourceName ||
-      newProps.project !== this.props.project
-    ) {
-      this.setState(this._loadFrom(newProps));
-    }
-  }
-
-  _loadFrom(props: Props): State {
-    const { project, resourceName } = props;
-    const resourcesManager = project.getResourcesManager();
-    const resourceKind = resourcesManager.hasResource(resourceName)
-      ? resourcesManager.getResource(resourceName).getKind()
-      : null;
+  React.useEffect(
+    () => {
+      const resourcesManager = project.getResourcesManager();
+      const resourceKind = resourcesManager.hasResource(resourceName)
+        ? resourcesManager.getResource(resourceName).getKind()
+        : null;
+      setResourceKind(resourceKind);
+    },
+    [project, resourceName]
+  );
 
-    return {
-      resourceKind,
-    };
+  switch (resourceKind) {
+    case null: // If no resource loaded yet, use the image thumbnail checkered background.
```

Comment:
I think the parent should pass the expected `resourceKind`. So we can always display what we know will be the type. And we just double check that the existing resource has the proper kind.

Reply:
In a sense, it's not normal that the child allow to display anything, it should only display what the parent know is right.

Reply:
yep, I'm passing down resourceKind now.
Not sure why it had its own logic there 🤔